subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c

```diff
@@ -292,7 +292,7 @@ static int write_resource_to_settings(int inst, int res, uint8_t *data, uint16_t
 	char path[sizeof(LWM2M_FIRM_PREFIX "/65535/0/10")];
 
 	snprintk(path, sizeof(path), LWM2M_FIRM_PREFIX "/%d/%d/%d", ENABLED_LWM2M_FIRMWARE_OBJECT,
-		 inst, res);
+		 MIN(inst, 999), MIN(res, 999));
```

Comment:
What is the point of this?
If the resource or instance is over 999, we end up writing garbage settings.

Why not just return failure or values are over certain limit?

Reply:
There are several ways to fix this I'm sure, whatever you prefer really. The buffer can also be increased.
This is the log warning:

```
nd 7 [-Wformat-truncation=]
   43 | #define LWM2M_FIRM_PREFIX "lwm2m:fir"
      |                           ^~~~~~~~~~~
/home/sir1/dev/ncs/zephyr/include/zephyr/sys/printk.h:67:32: note: in definition of macro 'snprintk'
   67 | #define snprintk(...) snprintf(__VA_ARGS__)
      |                                ^~~~~~~~~~~
/home/sir1/dev/ncs/nrf/subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:294:38: note: in expansion of macro 'LWM2M_FIRM_PREFIX'
  294 |         snprintk(path, sizeof(path), LWM2M_FIRM_PREFIX "/%d/%d/%d", ENABLED_LWM2M_FIRMWARE_OBJECT,
      |                                      ^~~~~~~~~~~~~~~~~
/home/sir1/dev/ncs/nrf/subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:1136:52: note: format string is defined here
 1136 |         char path[sizeof(LWM2M_FIRM_PREFIX "65632/0/10")];
      |                                                    ^~
/home/sir1/dev/ncs/nrf/subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:43:27: note: directive argument in the range [0, 65535]
   43 | #define LWM2M_FIRM_PREFIX "lwm2m:fir"
      |                           ^~~~~~~~~~~
/home/sir1/dev/ncs/zephyr/include/zephyr/sys/printk.h:67:32: note: in definition of macro 'snprintk'
   67 | #define snprintk(...) snprintf(__VA_ARGS__)
      |                                ^~~~~~~~~~~
/home/sir1/dev/ncs/nrf/subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:294:38: note: in expansion of macro 'LWM2M_FIRM_PREFIX'
  294 |         snprintk(path, sizeof(path), LWM2M_FIRM_PREFIX "/%d/%d/%d", ENABLED_LWM2M_FIRMWARE_OBJECT,
      |                                      ^~~~~~~~~~~~~~~~~
/home/sir1/dev/ncs/zephyr/include/zephyr/sys/printk.h:67:23: note: 'snprintf' output between 16 and 24 bytes into a destination of size 21
   67 | #define snprintk(...) snprintf(__VA_ARGS__)
      |                       ^~~~~~~~~~~~~~~~~~~~~
/home/sir1/dev/ncs/nrf/subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:294:9: note: in expansion of macro 'snprintk'
  294 |         snprintk(path, sizeof(path), LWM2M_FIRM_PREFIX "/%d/%d/%d", ENABLED_LWM2M_FIRMWARE_OBJECT,
```

The problem is that if the integer input variables are large the buffer `path` will not be large enough.

Reply:
Add
 ```c
if ((inst < 0 || inst > 9) || (res < 0 || res > 99)) {
	return -EINVAL;
}
```

Before the `sprintf()`. Compiler is clever enough to see now that the values are capped.

Reply:
thanks

Reply:
@SeppoTakalo added